pkg/proxy/winkernel/hcnutils.go

```diff
@@ -29,15 +29,51 @@ import (
 	"strings"
 )
 
-type hnsV2 struct{}
+type HCN interface {
+	GetNetworkByName(networkName string) (*hcn.HostComputeNetwork, error)
+	ListEndpointsOfNetwork(networkId string) ([]hcn.HostComputeEndpoint, error)
+	GetEndpointByID(endpointId string) (*hcn.HostComputeEndpoint, error)
+	ListEndpoints() ([]hcn.HostComputeEndpoint, error)
+	GetEndpointByName(endpointName string) (*hcn.HostComputeEndpoint, error)
+	ListLoadBalancers() ([]hcn.HostComputeLoadBalancer, error)
+	GetLoadBalancerByID(loadBalancerId string) (*hcn.HostComputeLoadBalancer, error)
+	CreateEndpoint(endpoint *hcn.HostComputeEndpoint, network *hcn.HostComputeNetwork) (*hcn.HostComputeEndpoint, error)
+	CreateLoadBalancer(loadbalancer *hcn.HostComputeLoadBalancer) (*hcn.HostComputeLoadBalancer, error)
+	CreateRemoteEndpoint(endpoint *hcn.HostComputeEndpoint, network *hcn.HostComputeNetwork) (*hcn.HostComputeEndpoint, error)
+	DeleteLoadBalancer(loadbalancer *hcn.HostComputeLoadBalancer) error
+	DeleteEndpoint(endpoint *hcn.HostComputeEndpoint) error
+}
+
+type ihcn struct{}
```

Comment:
Is the reason because of hns_test.go? If so, we should just remove the real HNS calls from hns_test.go and use the fake HCN in all the tests, if at all possible

Reply:
The ihcn actually represents the real hcn, so we can call the real hcn.methods whenever we need them. All it does is to call the real hcn methods. There is a single HCN interface, and fakeHCN and ihcn implementing that interface.  hcnutils has the hcninstance field that tells it to use ihcn (the real hcn implementation) or fakeHCN when calling the hcn methods that we use to implement the functions from  HCNUtils interface. It is not only for hns_test.go, but also for when we need to use hcnutils with the real hcn, not the fake one.

Reply:
This naming convention is confusing. The interface name's generally start with I, so may be IHCN could be the name of the interface with FakeHCN and RealHCN implementing IHCN.

Reply:
Interfaces in go typically don't have an `I` prepending them.